### DISCUSSION TITLE
Installing libMesh into specified location

### DISCUSSION THREAD

**Discussion by lloydgoodman** [idaholab/moose#18054] [Q&A Getting Started] [answered]
In the instructions for the hpc installation of moose (https://mooseframework.inl.gov/getting_started/installation/hpc_install_moose.html) there is mention of how to install PETSc to a specified location and then setting the PETSC_DIR variable.

Is it possible to do the same for libMesh?  Or is there some reason why you wouldn't want to or its not necessary?

**Comment by cticenhour** [comment, marked as answer]
This is possible. We track pretty closely with libMesh, so there is a danger of straying too far in version and having failures, but for development purposes you can build your own libmesh and set the `LIBMESH_DIR` environment variable to tell MOOSE where it's located. If you aren't doing active libmesh development, you should consider remaining with the submodule version supplied with MOOSE.
